### PR TITLE
[JENKINS-33596] Delete more of `UDPBroadcastThread` & `DNSMultiCast`

### PR DESCRIPTION
**Pull request by jglick**
Follows up #4460. See [JENKINS-33596](https://issues.jenkins-ci.org/browse/JENKINS-33596). These classes have been stubs for over a year and a half; it is just confusing to leave them in the source tree.

### Proposed changelog entries

* Removing long-unused class `UDPBroadcastThread`, and deleting more of `DNSMultiCast`.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by StefanSpieker — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Review by daniel-beck — APPROVED** [review]

**Comment by jglick** [comment]
Note that linked PRs mean that you will need to update tools to work with the new `jenkins.version`; you should be able to update tools anyway, though if you have a very old `jenkins.version` (< 2.220) you may see some delays and/or warnings.

**Comment by jglick** [comment]
Hmm, it occurs to me this could cause problems for people running PCT across a lot of plugins with older POMs. Maybe I need to keep just `DNSMultiCast.disabled` around to avoid linkage errors.

**Review by jeffret-b — APPROVED** [review]
I'm in favor of using these old things, but I do know from experience there can be some odd dependencies on the existence of some of these pieces. While I'd like to get rid of it all, a more cautious excision as this has evolved into is more prudent.

**Comment by jglick** [comment]
> a more cautious excision as this has evolved into is more prudent

Not sure I follow what you are saying. You approved this PR; do you have reservations?

**Comment by jeffret-b** [comment]
No reservations. And I think it's safer to leave a little bit of it behind like you ended up doing.

**Comment by jglick** [comment]
To be clear, we cannot safely delete `DNSMultiCast` now; we could only delete it after any plugin which you might wish to run PCT against has updated to a new POM.

File: test/pom.xml
Comment (timja) [inline test/pom.xml:84]: are the test failures related to updating this?

File: core/src/main/java/hudson/DNSMultiCast.java
Comment (jglick) [inline core/src/main/java/hudson/DNSMultiCast.java:7]: FTR tests still run with the older `JenkinsRule`

```diff
diff --git test/pom.xml test/pom.xml
index 1527fbcdcc..38d5934b15 100644
--- test/pom.xml
+++ test/pom.xml
@@ -81,7 +81,7 @@ THE SOFTWARE.
     <dependency>
       <groupId>${project.groupId}</groupId>
       <artifactId>jenkins-test-harness</artifactId>
-      <version>1626.v46b0925e70db</version>
+      <version>1589.vc23fca066d5c</version>
       <scope>test</scope>
       <exclusions>
         <exclusion>
```

File: test/pom.xml
Comment (timja) [reply test/pom.xml:84]: any idea why dependabot isn't updating this?

```
updater | INFO <job_213087289> Checking if org.jenkins-ci.main:jenkins-test-harness 1589.vc23fca066d5c needs updating
  proxy | 2021/09/26 19:10:19 [146] GET https://repo.jenkins-ci.org:443/public/org/jenkins-ci/jenkins/1.67/jenkins-1.67.pom
  proxy | 2021/09/26 19:10:19 [146] 200 https://repo.jenkins-ci.org:443/public/org/jenkins-ci/jenkins/1.67/jenkins-1.67.pom
  proxy | 2021/09/26 19:10:19 [148] GET https://repo.jenkins-ci.org:443/public/org/jenkins-ci/main/jenkins-test-harness/maven-metadata.xml
  proxy | 2021/09/26 19:10:19 [148] 200 https://repo.jenkins-ci.org:443/public/org/jenkins-ci/main/jenkins-test-harness/maven-metadata.xml
  proxy | 2021/09/26 19:10:19 [150] GET https://repo.maven.apache.org:443/maven2/org/jenkins-ci/main/jenkins-test-harness/maven-metadata.xml
  proxy | 2021/09/26 19:10:19 [150] 404 https://repo.maven.apache.org:443/maven2/org/jenkins-ci/main/jenkins-test-harness/maven-metadata.xml
  proxy | 2021/09/26 19:10:19 [152] HEAD https://repo.jenkins-ci.org:443/public/org/jenkins-ci/main/jenkins-test-harness/1589.vc23fca066d5c/jenkins-test-harness-1589.vc23fca066d5c.jar
  proxy | 2021/09/26 19:10:19 [152] 200 https://repo.jenkins-ci.org:443/public/org/jenkins-ci/main/jenkins-test-harness/1589.vc23fca066d5c/jenkins-test-harness-1589.vc23fca066d5c.jar
updater | INFO <job_213087289> Latest version is 1589.vc23fca066d5c
```

Comment (jglick) [reply test/pom.xml:84]: No idea offhand.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**
